### PR TITLE
Remove bpo-27931 workaround

### PR DESCRIPTION
**Pull request by pepoluan**
## What do these changes do?

[**bpo-27931**](https://bugs.python.org/issue27931) is a bug in Python's email address parser. It has been fixed since Python 3.6.

Since we have dropped support for Python < 3.6, it serves that we should no longer test for this workaround.

Rather, we test Python>=3.6 email address parser against a list of "example" valid & invalid addresses.

This is partially adopted from PR #202 which had implemented the necessary changes, but still quite some way from being merge-able.

## Are there changes in behavior for the user?

None for server-side implementors.

Clients _might_ have to adjust because the vocabulary for responses have been expanded (addition of `553 5.3.1 Error: malformed address` error code)

## Related issue number

Can't find any.

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] tox testenvs have been executed in the following environments:
   - [x] Windows 10 (via PyCharm tox runner): `(ALL)`
   - [x] Windows 10 (via PSCore 7.1.0): `(ALL)`
   - [x] Windows 10 (via Cygwin): `qa,py36-{nocov,cov}`
   - [x] Ubuntu 18.04 on WSL 1.0: `(ALL)` + `pypy3-{nocov,cov,diffcov}`
   - [x] Ubuntu 18.04 on VBox: `(ALL)` + `pypy3-{nocov,cov,diffcov}`
   - [x] Ubuntu 20.04 on VBox: `(ALL)` + `pypy3-{nocov,cov,diffcov}`
   - [x] FreeBSD 12.1 on VBox: `(ALL)` + `pypy3-nocov`
- [x] Documentation reflects the changes _-- not really applicable since change is not user-visible_
- [x] Add a news fragment into the `NEWS.rst` file

**_This post is undergoing edits as I finish up on testing_**

### CODE REVIEW

File: aiosmtpd/smtp.py [hunk 1]
```diff
@@ -928,9 +936,10 @@ async def smtp_RCPT(self, arg: str) -> None:
             await self.push(syntaxerr)
             return
         address, params = self._getaddr(arg)
+        if address is None:
+            return await self.push("553 5.1.3 Error: malformed address")
```
Comment (warsaw) [inline aiosmtpd/smtp.py:940]: Since you're using this string twice, maybe refactor it into a constant?

File: aiosmtpd/smtp.py [hunk 2]
```diff
@@ -928,9 +936,10 @@ async def smtp_RCPT(self, arg: str) -> None:
...
         if not address:
-            await self.push(syntaxerr)
-            return
+            return await self.push(syntaxerr)
         if not self.session.extended_smtp and params:
             await self.push(syntaxerr)
             return
```
Comment (warsaw) [inline aiosmtpd/smtp.py:945]: Maybe make this style match the above (line 942)?

**Review by warsaw — APPROVED** [review]
A few minor comments, but otherwise LGTM!

File: aiosmtpd/smtp.py (hunk 1 above)
Comment (pepoluan) [reply aiosmtpd/smtp.py:940]: In essence, I agree. But I do have the idea of making _all_ SMTP status code string into a class of messages, similar (but not identical) to [this file](https://github.com/pepoluan/aiosmtpd/blob/pytest_based/aiosmtpd/testing/statuscodes.py) (part of PR #202 ).

Part of the reason is to make it easier to change if necessary (no need to hunt through all `smtp.py`), and some thoughts about i18n (though I honestly can't find info if SMTP status messages are allowed in non-English...). Maybe by making it into some sort of "pluggable" class, similar to handler?

File: aiosmtpd/smtp.py (hunk 2 above)
Comment (pepoluan) [reply aiosmtpd/smtp.py:945]: That's an oopsie 😅 

Let me do some styling edits + commit + push.

**Comment by pepoluan** [comment]
Since this is for milestone 1.3 anyways, I'll let this PR 'marinate' for awhile, say 4x24, before squamerging.

**Comment by pepoluan** [comment]
I'll give additional 24h before squamerging.

**Pull request merged**